src/utils.test.ts

```diff
@@ -39,6 +42,34 @@ describe("convertSkylinkToBase32", () => {
   });
 });
 
+describe("getRelativeFilePath", () => {
+  // TODO: Is this right?
+  const filepaths = [
+    ["abc/def", "abc:def"],
+    ["./abc/def", ".:abc:def"],
+    ["/abc/def.txt", ":abc:def.txt"],
+  ];
+
+  it.each(filepaths)("the relative file path for a file with the path %s should be %s", (filepath, directory) => {
+    const file = new File(["test contents"], filepath);
+    expect(getRelativeFilePath(file)).toEqual(directory);
+  });
+});
+
+describe("getRootDirectory", () => {
+  // TODO: Is this right?
+  const filepaths = [
+    ["abc/def", "."],
+    ["./abc/def", "."],
+    ["/abc/def", "."],
+  ];
+
+  it.each(filepaths)("the root directory for a file with the path %s should be %s", (filepath, directory) => {
+    const file = new File(["test contents"], filepath);
```

Comment:
```suggestion
    const file = new File(["test contents"], "foo");
    file.path = filepath;
```

You need to spoof the `file.path` prop because if you provide the file path as a name like you did, it will escape all `/` into `:` and it will not detect the root dir properly.